edb/lib/std/25-booloperators.edgeql

```diff
@@ -34,70 +34,96 @@
 # when inside a WHERE clause, so we must use the original
 # boolean expression in conjunction.
 CREATE INFIX OPERATOR
-std::`OR` (a: std::bool, b: std::bool) -> std::bool
+std::`OR` (a: std::bool, b: std::bool) -> std::bool {
+    SET volatility := 'IMMUTABLE';
     FROM SQL $$
     SELECT ("a" OR "b") AND ("a"::int | "b"::int)::bool
     $$;
+};
 
 
 # `FROM SQL EXPRESSION` means that the operator is translated
 # by the compiler into some SQL expression.
 CREATE INFIX OPERATOR
-std::`AND` (a: std::bool, b: std::bool) -> std::bool
+std::`AND` (a: std::bool, b: std::bool) -> std::bool {
+    SET volatility := 'IMMUTABLE';
     FROM SQL EXPRESSION;
+};
 
 
 CREATE PREFIX OPERATOR
-std::`NOT` (v: std::bool) -> std::bool
+std::`NOT` (v: std::bool) -> std::bool {
+    SET volatility := 'IMMUTABLE';
     FROM SQL EXPRESSION;
+};
 
 
 CREATE INFIX OPERATOR
-std::`=` (l: std::bool, r: std::bool) -> std::bool
+std::`=` (l: std::bool, r: std::bool) -> std::bool {
+    SET volatility := 'IMMUTABLE';
     FROM SQL OPERATOR r'=';
+};
 
 
 CREATE INFIX OPERATOR
-std::`?=` (l: OPTIONAL std::bool, r: OPTIONAL std::bool) -> std::bool
+std::`?=` (l: OPTIONAL std::bool, r: OPTIONAL std::bool) -> std::bool {
+    SET volatility := 'IMMUTABLE';
     FROM SQL EXPRESSION;
+};
 
 
 CREATE INFIX OPERATOR
-std::`!=` (l: std::bool, r: std::bool) -> std::bool
+std::`!=` (l: std::bool, r: std::bool) -> std::bool {
+    SET volatility := 'IMMUTABLE';
     FROM SQL OPERATOR r'<>';
+};
 
 
 CREATE INFIX OPERATOR
-std::`?!=` (l: OPTIONAL std::bool, r: OPTIONAL std::bool) -> std::bool
+std::`?!=` (l: OPTIONAL std::bool, r: OPTIONAL std::bool) -> std::bool {
+    SET volatility := 'IMMUTABLE';
     FROM SQL EXPRESSION;
+};
 
 
 CREATE INFIX OPERATOR
-std::`>=` (l: std::bool, r: std::bool) -> std::bool
+std::`>=` (l: std::bool, r: std::bool) -> std::bool {
+    SET volatility := 'IMMUTABLE';
```

Comment:
You set a default volatility for all operators to `IMMUTABLE`. Why this duplication?

Reply:
Somewhat confusingly, he set a default on the *introspection* type, the overall default is `VOLATILE`:

https://github.com/edgedb/edgedb/pull/661/files#diff-de65a2723408cb4f1ab36bd8fed914e1R402

Reply:
* What's the point of setting it on the introspection type? 
* Why is it different? 
* Should we set the default for operators to IMMUTABLE?

Reply:
We should't be setting defaults on introspection types.  Operators aren't special (but they tend to be immutable).

Reply:
I've removed the defaults for volatility except the one for the base class. I left it in mainly for purposes of showing that there IS a default value, which otherwise would remain buried in python.